Add merged to downloadable files (#234)
AlexsLemonade/scpca-docs: docs/download_files.md

Note that if a sample selected for download contains a spatial transcriptomics library, the files included will be different than pictured below.
See the [description of the Spatial transcriptomics output section below](#spatial-transcriptomics-libraries).

### `SingleCellExperiment` downloads

### Download folder structure for project downloads:
![project download folder](images/project-download-folder.png)

### Download folder structure for individual sample downloads:
![sample download folder](images/sample-download-folder.png)

### `AnnData` downloads

#### Download folder structure for project downloads:
![project download folder](images/anndata-project-download-folder.png)

#### Download folder structure for individual sample downloads:
![sample download folder](images/anndata-sample-download-folder.png)

#### Download folder structure for individual sample downloads with CITE-seq (ADT) data:
![sample download folder](images/anndata-sample-citeseq-download-folder.png)

If downloading a sample that contains a CITE-seq library as an `AnnData` object (`hdf5` file), the quantified CITE-seq expression data is included as a separate file.

> 

Reviewer: I think I get where you are going here, but I would move the merged object section to be immediately after all the download descriptions. It should be on the same level as `Gene expression data`, `QC report` etc. All of that is the same for both types of downloads. The only additional thing is the merge report which you can address in the text describing downloading projects as merged objects.

Reviewer: Yea I see what you are saying. I think I would then leave everything at the level they were originally add, but then treat the merged object download as a special consideration section similar to spatial and multiplexed. So it should be on the same level as those items. Maybe after the multiplexed section and before the spatial section?

Author: > So it should be on the same level as those items. Maybe after the multiplexed section and before the spatial section?

Yeah, I'm good with this!

The [metadata file](#metadata) for multiplexed libraries (`single_cell_metadata.tsv`) will have the same format as for individual samples, but each row will represent a particular sample/library pair, meaning that there may be multiple rows for each `scpca_library_id`, one for each `scpca_sample_id` within that library.

## Merged object downloads

> Merged object downloads contain all single-cell or single-nuclei gene expression data for a given ScPCA project within a single object, provided as either a [`SingleCellExperiment` object (`.rds` file)](http://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html) or an [`AnnData` object (`.hdf5` file)](https://anndata.readthedocs.io/en/latest/index.html).

Reviewer: I would add a sentence here first that says something like: 

"If downloading all samples associated with a project, data can be downloaded so that each sample is in an individual file or all data is in a single file or merged object."

I don't think the wording is quite right, but this is the message I would portray so people understand what use case they would even want a merged object.

The object file, `SCPCP000000_merged.rds` or `SCPCP000000_merged_rna.hdf5`, contains both a raw and normalized counts matrix, each with combined counts for all libraries in an ScPCA project.
In addition to the counts matrices, the `SingleCellExperiment` or `AnnData` object stored in the file includes the results of library-weighted dimensionality reduction using both principal component analysis (PCA) and UMAP.
See the {ref}`section on merged object processing<processing_information:merged objects>` for more information about how merged objects were created.

> If downloading a project that contains at least one CITE-seq library as an `AnnData` object (`hdf5` file), the quantified CITE-seq expression data is included as a separate file, `SCPCP000000_merged_adt.hdf5`.

Reviewer: I think I would mention that the data is included in the same file for SCE, but for AnnData it's different rather than just mention one.

> Every download includes summary report, `SCPCL000000_merged-summary-report.html`, which provides brief information about each library, including technology and sample diagnoses, as well as UMAP visualizations highlighting each library.

Reviewer: ```suggestion
Every download includes a summary report, `SCPCL000000_merged-summary-report.html`, which provides a 
 a brief summary of the libraries included in the merged object.
This includes a summary of the types of libraries (e.g., single-cell, single-nuclei, with CITE-seq) and sample diagnoses included in the object, as well as UMAP visualizations highlighting each library.
```
Since technically there's not information about each library, we should emphasize that it's a summary.

Every download also includes the individual [QC report](#qc-report) and, if applicable, [cell type annotation reports](#cell-type-report) for each library included in the merged object.

### Download folder structure for `SingleCellExperiment` merged downloads:
_image pending_

### Download folder structure for `AnnData` merged downloads:
_image pending_

### Download folder structure for `AnnData` merged downloads with CITE-seq (ADT) data:

> _image pending_

Reviewer: I think as long as we file an issue for the images and note the changes that we are waiting on, then this can wait for another PR.

### Merged object metadata

> Every download includes a single `single_cell_metadata.tsv` file containing metadata for all libraries included in the download.

Reviewer: ```suggestion
Similar to downloading the project with individual files for each sample, downloading the project as a merged object includes a single `single_cell_metadata.tsv` file containing metadata for all libraries included in the download.
```

Every download also includes the individual [QC report](#qc-report) and, if applicable, [cell type annotation reports](#cell-type-report) for each library included in the merged object.

### Download folder structure for `SingleCellExperiment` merged downloads:
_image pending_

### Download folder structure for `AnnData` merged downloads:
_image pending_

### Download folder structure for `AnnData` merged downloads with CITE-seq (ADT) data:
_image pending_

### Merged object metadata

Similar to downloading the project with individual files for each sample, downloading the project as a merged object includes a single `single_cell_metadata.tsv` file containing metadata for all libraries included in the download.

If downloading a project containing bulk RNA-seq data, two tab-separated value files, `bulk_quant.tsv` and `bulk_metadata.tsv`, will be included in the project download.
The `bulk_quant.tsv` file contains a gene by sample matrix (each row a gene, each column a sample) containing raw gene expression counts quantified by Salmon.

> The `bulk_metadata.tsv` file contains associated metadata for all samples with bulk RNA-seq data.

Reviewer: I think I might move this section to be right below the images or right above. Putting it right below would mirror how we mention bulk for the single sample downloads.

Author: Do you mean the bulk text specifically or the little section for merged metadata downloads? Currently this is right below the images.
![Screenshot 2024-01-22 at 1 48 37 PM](https://github.com/AlexsLemonade/scpca-docs/assets/4701111/be9357e4-e999-4c7c-9fa0-a9ece4a26ea5)

Reviewer: I mean to move the `If downloading a project containing bulk RNA-seq` and the three lines that mention bulk to be in the section that actually shows the download images and remove from the header `Merged object metadata`.

The [metadata file](#metadata) for multiplexed libraries (`single_cell_metadata.tsv`) will have the same format as for individual samples, but each row will represent a particular sample/library pair, meaning that there may be multiple rows for each `scpca_library_id`, one for each `scpca_sample_id` within that library.

## Merged object downloads

> When downloading a full ScPCA project, you can choose to download each library as an individual file, or you can download {ref}`a single file containing all libraries merged into a single object<faq:INCOMING MERGED OBJECT SECTION>`. TODO!

Reviewer: ```suggestion
When downloading a full ScPCA project, you can choose to download data from all samples as individual files, or you can download {ref}`a single file containing all libraries from all samples merged into a single object<faq:INCOMING MERGED OBJECT SECTION>`. TODO!
```
Copying some of my thoughts from my review in #244, we should use samples in the language.

The object file, `SCPCP000000_merged.rds` or `SCPCP000000_merged_rna.hdf5`, contains both a raw and normalized counts matrix, each with combined counts for all libraries in an ScPCA project.
In addition to the counts matrices, the `SingleCellExperiment` or `AnnData` object stored in the file includes the results of library-weighted dimensionality reduction using both principal component analysis (PCA) and UMAP.
See the {ref}`section on merged object processing<processing_information:merged objects>` for more information about how merged objects were created.

If downloading a project that contains at least one CITE-seq library, the quantified CITE-seq expression data will also be merged.
In `SingleCellExperiment` objects (`rds` files), the  CITE-seq expression data is be provided as an alternative experiment in the same object as the gene expression data.
However, for `AnnData` objects, (`hdf5` files), the quantified CITE-seq expression is instead provided as a separate file called `SCPCP000000_merged_adt.hdf5`.

> Every download includes a summary report, `SCPCL000000_merged-summary-report.html`, which provides a
>  a brief summary of the libraries included in the merged object.

Reviewer: ```suggestion
Every download includes a summary report, `SCPCL000000_merged-summary-report.html`, which provides a brief summary of the samples and libraries included in the merged object.
```

Every download also includes the individual [QC report](#qc-report) and, if applicable, [cell type annotation reports](#cell-type-report) for each library included in the merged object.

### Download folder structure for `SingleCellExperiment` merged downloads:
_image pending_

### Download folder structure for `AnnData` merged downloads:
_image pending_

### Download folder structure for `AnnData` merged downloads with CITE-seq (ADT) data:
_image pending_

> 
> 
> ### Merged object metadata
> 
> If downloading a project containing bulk RNA-seq data, two tab-separated value files, `bulk_quant.tsv` and `bulk_metadata.tsv`, will be included in the project download.
> The `bulk_quant.tsv` file contains a gene by sample matrix (each row a gene, each column a sample) containing raw gene expression counts quantified by Salmon.
> The `bulk_metadata.tsv` file contains associated metadata for all samples with bulk RNA-seq data.

Reviewer: ```suggestion

If downloading a project containing bulk RNA-seq data, two tab-separated value files, `bulk_quant.tsv` and `bulk_metadata.tsv`, will be included in the project download.
The `bulk_quant.tsv` file contains a gene by sample matrix (each row a gene, each column a sample) containing raw gene expression counts quantified by Salmon.
The `bulk_metadata.tsv` file contains associated metadata for all samples with bulk RNA-seq data.

### Merged object metadata
```

> Similar to downloading the project with individual files for each sample, downloading the project as a merged object includes a single `single_cell_metadata.tsv` file containing metadata for all samples included in the download.

Reviewer: Maybe we should link to the other metadata section and mention that a full table is there with the expected contents of the metadata file?

Author: good idea!

Author: Done: https://github.com/AlexsLemonade/scpca-docs/pull/234/commits/ee8f549db1503c7e289d39aaf7c449e7a83ba891

Merged object downloads contain all single-cell or single-nuclei gene expression data for a given ScPCA project within a single object, provided as either a [`SingleCellExperiment` object (`.rds` file)](http://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html) or an [`AnnData` object (`.hdf5` file)](https://anndata.readthedocs.io/en/latest/index.html).

The object file, `SCPCP000000_merged.rds` or `SCPCP000000_merged_rna.hdf5`, contains both a raw and normalized counts matrix, each with combined counts for all samples in an ScPCA project.
In addition to the counts matrices, the `SingleCellExperiment` or `AnnData` object stored in the file includes the results of library-weighted dimensionality reduction using both principal component analysis (PCA) and UMAP.
See the {ref}`section on merged object processing<processing_information:merged objects>` for more information about how merged objects were created.

If downloading a project that contains at least one CITE-seq library, the quantified CITE-seq expression data will also be merged.

> In `SingleCellExperiment` objects (`rds` files), the  CITE-seq expression data is be provided as an alternative experiment in the same object as the gene expression data.

Reviewer: ```suggestion
In `SingleCellExperiment` objects (`rds` files), the  CITE-seq expression data is provided as an alternative experiment in the same object as the gene expression data.
```
